controlplane/kubeadm/internal/controllers/status.go

```diff
@@ -108,6 +108,11 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, contro
 	if status.HasKubeadmConfig {
 		controlPlane.KCP.Status.Initialized = true
 		conditions.MarkTrue(controlPlane.KCP, controlplanev1.AvailableCondition)
+		v1beta2conditions.Set(controlPlane.KCP, metav1.Condition{
+			Type:   controlplanev1.KubeadmControlPlaneInitializedV1Beta2Condition,
+			Status: metav1.ConditionTrue,
+			Reason: controlplanev1.KubeadmControlPlaneInitializedV1Beta2Reason,
+		})
```

Comment:
What about moving this (and also when we set the condition to false) into a a new setInitializedCondition func called by updateV1beta2Status.

updateV1beta2Status is always called after updateStatus, and so we can test controlPlane.KCP.Status.Initialized and set the condition accordingly; this will prevent mixing up V1beta2 conditions into update status